lib/bitbucket_rest_api/request/oauth2.rb

```diff
@@ -0,0 +1,54 @@
+require 'base64'
+require 'faraday'
+require 'forwardable'
+
+module BitBucket
+  module FaradayMiddleware
+    class OAuth2 < Faraday::Middleware
+      AUTH_HEADER = 'Authorization'.freeze
+
+      attr_reader :options
+
+      extend Forwardable
+
+      def call(env)
+        reset_oauth2_token! if oauth2_token.expired?
```

Comment:
We are using client credentials which is not suppose to return a refresh token. According to the spec you should just re-retrieve the access token using client credentials.